gravitee-node-secrets/gravitee-secret-provider-mock/src/main/java/io/gravitee/node/secrets/plugin/mock/MockSecretProvider.java

```diff
@@ -0,0 +1,121 @@
+                    throw new MockSecretProviderException(message);
+                }
+            }
+
+            if (renewalsReturned.containsKey(location.secret())) {
+                AtomicInteger counter = renewalsReturned.get(location.secret());
+                Renewal renewal = configuration.getConfiguredRenewals().get(location.secret());
+                if (counter.get() < renewal.revisions().size()) {
+                    // next revision and increment
+                    return getAndIncrement(renewal, counter);
+                } else if (renewal.loop()) {
+                    // went over just reset
+                    counter.set(0);
+                } else {
+                    // went over no looping => get latest, simulate that won't change anymore
+                    return SecretMap.of(renewal.revisions().get(counter.get() - 1));
+                }
+            }
+            if (configuration.getConfiguredRenewals().containsKey(location.secret())) {
+                renewalsReturned.put(location.secret(), new AtomicInteger());
+            }
+
+            // standard resolution
+            Map<String, Object> secretMap = ConfigHelper.removePrefix(configuration.getSecrets(), location.secret());
+            if (secretMap.isEmpty()) {
+                log.info("{}-{} no secrets for: {}", PLUGIN_ID, SECRET_PROVIDER, location.secret());
+                return null;
+            }
+            log.info("{}-{} found secrets ({}) for: {}", PLUGIN_ID, SECRET_PROVIDER, secretMap.size(), location.secret());
+            return SecretMap.of(secretMap);
+        });
+    }
+
+    private static boolean shouldReturnRaiseError(ConfiguredError error, AtomicInteger errorReturned) {
+        return error.repeat() == 0 || error.repeat() > 0 && errorReturned.getAndIncrement() < error.repeat();
+    }
+
+    private static SecretMap getAndIncrement(Renewal renewal, AtomicInteger counter) {
+        return SecretMap.of(renewal.revisions().get(counter.getAndIncrement()));
+    }
+
+    @Override
+    public Flowable<SecretEvent> watch(SecretMount secretMount) {
+        MockSecretLocation location = MockSecretLocation.fromLocation(secretMount.location());
+        List<ConfiguredEvent> list = configuration
+            .getConfiguredEvents()
+            .stream()
+            .filter(e -> e.secret().equals(location.secret()))
+            .toList();
+        return Flowable
+            .fromIterable(list)
+            .delay(configuration.getWatchesDelayDuration(), configuration.getWatchesDelayUnit())
+            .flatMapSingle(event -> {
+                if (Objects.equals(event.error(), "null")) {
+                    return Single.just(new SecretEvent(event.type(), SecretMap.of(event.data())));
+                } else {
+                    return Single.error(new MockSecretProviderException(event.error()));
+                }
+            });
+    }
+
+    @Override
+    public SecretMount fromURL(SecretURL url) {
```

Comment:
Should you check the url ?

Reply:
It is usually parse from a string, so no I don't, but I will